tsdb/engine/tsm1/file_store.go

```diff
@@ -633,6 +629,43 @@ func (f *FileStore) BlockCount(path string, idx int) int {
 	return 0
 }
 
+// walkFiles calls fn for every files in filestore in parallel
+func (f *FileStore) walkFiles(fn func(f TSMFile) error) error {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	// struct to hold the result of opening each reader in a goroutine
+	type res struct {
+		err error
+	}
+
+	resC := make(chan res)
+	var n int
```

Comment:
nit: this is just a style thing, but since you know you're sending a result for each file, couldn't you get rid of `n` completely and just create a buffered channel of length `len(f.files)`? 

In your gathering loop at the bottom of `walkFiles`, you would then loop `len(resC)` times.